packages/datetime/src/dateRangeInput.tsx

```diff
@@ -180,6 +186,14 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
         };
     }
 
+    public componentDidUpdate() {
+        if (this.state.isStartInputFocused && document.activeElement !== this.startInputRef) {
+            this.startInputRef.focus();
+        } else if (this.state.isEndInputFocused && document.activeElement !== this.endInputRef) {
+            this.endInputRef.focus();
```

Comment:
Out of curiosity, when would that happen? When the component is being mounted/dismounted?